hydrolib/core/io/structure/models.py

```diff
@@ -62,15 +63,112 @@ class Comments(INIBasedModel.Comments):
     y_coordinates: Optional[List[float]] = Field(None, alias="yCoordinates")
 
     @root_validator
-    def check_location(cls, values):
+    @classmethod
+    def check_location(cls, values: dict) -> dict:
+        """
+        Validates the location of the structure based on the given parameters.
+        For instance, if a branchId is given, then it is expected also the chainage,
+        otherwise numCoordinates x and yCoordinates shall be expected.
+
+        Args:
+            values (dict): Dictionary of values validated for the new structure.
+
+        Raises:
+            ValueError: When branchid or chainage values are not valid (empty strings).
+            ValueError: When the number of coordinates (x,y) do not match numCoordinates.
+
+        Returns:
+            dict: Dictionary of values validated for the new structure.
+        """
+        # Verify this evaluation needs to be done.
+        filtered_values = {k: v for k, v in values.items() if v is not None}
+        structure_type = filtered_values.get("structure_type", "").lower()
+        if structure_type == "compound" or issubclass(cls, Compound):
+            # Compound structure does not require a location specification.
+            return values
+
+        coordinates_in_model = Structure.validate_coordinates_in_model(filtered_values)
+
+        # Exception -> LongCulvert requires coordinates_in_model, but not branchId and chainage.
+        if structure_type == "longculvert":
+            assert (
+                coordinates_in_model
+            ), "Coordinate system is mandatory for a LongCulvert structure."
```

Comment:
Terminology: "Coordinate system" is often used for geographical coordinate reference system descriptions (i.e., not the coordinates themselves). I'd use "num/x/yCoordinates are mandatory for a longCulvert structure."